pkg/network/setup/netstat_test.go

```diff
@@ -164,6 +164,34 @@ var _ = Describe("netstat", func() {
 
 			Expect(netStat.PodInterfaceVolatileDataIsCached(vmi, primaryNetworkName)).To(BeTrue())
 		})
+
+		It("should update existing interface status with MAC from the domain", func() {
+			const (
+				origMAC      = "C0:01:BE:E7:15:G0:0D"
+				newDomainMAC = "1C:CE:C0:01:BE:E7"
+			)
+
+			vmi.Status.Interfaces = []v1.VirtualMachineInstanceNetworkInterface{
+				{
+					IP:   primaryPodIPv4,
+					IPs:  []string{primaryPodIPv4, primaryPodIPv6},
+					MAC:  origMAC,
+					Name: primaryNetworkName,
+				},
+			}
+
+			domain := &api.Domain{
+				Spec: api.DomainSpec{Devices: api.Devices{Interfaces: []api.Interface{
+					newDomainSpecIface(primaryNetworkName, newDomainMAC),
+				}}},
+			}
+
+			netStat.UpdateStatus(vmi, domain)
+
+			Expect(vmi.Status.Interfaces).To(Equal([]v1.VirtualMachineInstanceNetworkInterface{
+				newVMIStatusIface(primaryNetworkName, []string{primaryPodIPv4, primaryPodIPv6}, newDomainMAC, ""),
```

Comment:
Now I've realized you're mixing up two separate tests (that are 99% equal) in a single one. 

Would it make sense to make this a table and pass these IPs as input parameters of the function ? Or is having a "single" more complex case enough ?

Reply:
I have not created a new test, just moved it to have the cache available to it (pod interface cache).
But what two tests you are talking about? This one was originally targeted to check that the MAC address is updated based on the domain information (i.e. it overrides the info collected from the pod or anything that existed in the status before, which is now dropped).